### PR TITLE
Add support for OEM and sending error events

### PR DESCRIPTION
**Pull request by marineam**

### CODE REVIEW

**Comment by marineam** [comment]
Found some issues with this, rework coming soon.

File: omaha/client/client.go [hunk 1]
```diff
@@ -222,36 +245,52 @@ func (ac *AppClient) Ping() error {
...
-
-	return nil
+	go func() {
+		appResp, err := ac.doReq(url, req)
+		if err != nil {
+			errc <- err
+			return
+		}
+
+		// BUG: CoreUpdate does not send event status in response.
```
Comment (yifan-gu) [inline omaha/client/client.go]: Did we create issue on CoreUpdate for this?

Comment (marineam) [inline omaha/client/client.go]: No, I haven't yet.

Comment (marineam) [inline omaha/client/client.go]: Issue submitted.

File: README.md [hunk 2]
```diff
@@ -14,7 +14,8 @@ These differences include:
  - No offline activity tracking.
    The protocol's ping mechanism allows for tracking application usage, reporting the number of days since the last ping and how many of those days saw active usage.
    CoreUpdate does not use this, instead assuming update clients are always online and checking in once every ~45-50 minutes.
-   Each check in should include a ping and optionally an update check.
+   Clients not actively updating should send only a ping, indicating CoreUpdate's "Instance-Hold" state.
+   Clients requesting an update should send a ping, update check, and an UpdateComplete:SuccessReboot event indicating CoreUpdate's "Complete" state.
```
Comment (yifan-gu) [inline README.md]: does tco need to send "SuccessReboot"?

Comment (marineam) [inline README.md]: It already does

File: omaha/client/client.go [hunk 3]
```diff
@@ -276,6 +325,23 @@ func (ac *AppClient) newReq() *omaha.Request {
...
+	} else if err, ok := err.(ErrorEvent); ok {
+		ac.Event(err.ErrorEvent())
+	} else if err != nil {
+		ac.Event(NewErrorEvent(ExitCodeOmahaRequestError))
+	}
+	return resp, err
+}
+
+// doReq posts an omaha request. It may be called in its own goroutine so
+// it should not touch any mutable data in AppClient. ac.apiClient is ok.
```
Comment (yifan-gu) [inline omaha/client/client.go]: Empty space after `AppClient.`?

Comment (marineam) [inline omaha/client/client.go]: That's the end of the sentence, but perhaps a coma is more readable.

Comment (marineam) [inline omaha/client/client.go]: Fixed in aea5c5b

File: omaha/client/client.go (hunk 1 above)
Comment (yifan-gu) [inline omaha/client/client.go]: nit: Maybe link the issue in the comments?
https://github.com/coreos-inc/roller/issues/750

**Comment by yifan-gu** [comment]
LGTM after small nits and squash.

**Comment by marineam** [comment]
Squashed

**Pull request merged**